File: bk-api/dao/bookmark_dao.go
package dao

import (
	"context"
	"github.com/mageddo/go-logging"
	"bk-api/entity"
	"database/sql"
)

type BookmarkDAO interface {
	LoadSiteMap() ([]entity.BookmarkEntity, error)
	GetBookmarks(offset, quantity int) ([]entity.BookmarkEntity, int, error)
	GetBookmarksByTagSlug(slug string, offset, quantity int) ([]entity.BookmarkEntity, int, error)
	GetBookmarksByNameOrHTML(query string, offset, quantity int) ([]entity.BookmarkEntity, int, error)
	SaveBookmark(tx *sql.Tx, bookmark *entity.BookmarkEntity) error
}

func NewBookmarkDAO(ctx context.Context) *BookmarkDAOSQLite {
	return &BookmarkDAOSQLite{logging.NewLog(ctx)}
}

File: src/main/java/com/mageddo/bookmarks/dao/BookmarkDAO.java
package com.mageddo.bookmarks.dao;

import java.util.List;

import com.mageddo.bookmarks.apiserver.res.BookmarkDescriptionRes;
import com.mageddo.bookmarks.apiserver.res.BookmarkRes;
import com.mageddo.bookmarks.apiserver.res.RecentBookmarksRes;
import com.mageddo.bookmarks.entity.BookmarkEntity;

public interface BookmarkDAO {

  void createBookmark(BookmarkEntity bookmarkEntity);

  List<BookmarkEntity> loadSiteMap();

  List<BookmarkRes> getBookmarks(String tag, int offset, int quantity);

  List<BookmarkRes> getBookmarksByNameOrContent(String query, int offset, int quantity);

  List<BookmarkRes> getBookmarks(int offset, int quantity);

  int countPublicNotDeleted();

  List<RecentBookmarksRes> getRecentBookmarks(int pageSize, int startPage);

  BookmarkRes findBookmarkRes(long bookmarkId);

  void update(BookmarkEntity bookmark);

  void associate(long tagId, long bookmarkId);

  void disassociateTags(long bookmarkId);

  BookmarkDescriptionRes findBookMarkWithNavigation(int bookmarkId);

  void deleteBookmark(int bookmarkId);

  void recoverBookmark(int bookmarkId);
}
